pool.go

```diff
@@ -125,12 +139,12 @@ type Pool[T any] struct {
 	acquireCount         int64
 	acquireDuration      time.Duration
 	emptyAcquireCount    int64
-	canceledAcquireCount int64
+	canceledAcquireCount atomic.Int64
```

Comment:
Is there any guarantee that this is 64-bit aligned on 32 bit systems? If not, you'll get panics when running on a 32 bit system. And you'll have to be very careful every time you change this struct. This is a constant source of annoying with int64 atomics.

See https://github.com/golang/go/issues/19057 for some context, and https://github.com/tailscale/tailscale/blob/main/types/pad32/pad32.go for the sort of shenanigans that might be useful.

Reply:
Structs are always 64 bit aligned. I elaborate on this topic in my previous comment: https://github.com/jackc/puddle/pull/24#issuecomment-1277363623

Reply:
Ah, I missed that the library wraps them in a struct.